### PR TITLE
Use yapf for formatting

### PR DESCRIPTION
**Pull request by sergiogiro**

### CODE REVIEW

File: services/controller/Pipfile [hunk 1]
```diff
@@ -14,3 +14,4 @@ requests = ">=2.20.0"
 
 [dev-packages]
 "flake8" = "*"
+yapf = "*"
```
Comment (SamirTalwar) [inline services/controller/Pipfile]: Why isn't this also a dependency for the CLI?

File: Makefile [hunk 2]
```diff
@@ -4,6 +4,7 @@ include vars.mk
 
 .PHONY: check
 check:
+	cd services/controller; pipenv install --dev; pipenv run yapf -rd ../..
```
Comment (SamirTalwar) [inline Makefile]: Make this part of the *cli* and *services/controller* Makefiles, and have them depend on the *environment* target.

Comment (SamirTalwar) [reply Makefile]: It'd be good to add a *format* target to *python.mk*, and have a root one that delegates to both.

File: cli/src/plz/cli/controller_proxy.py [hunk 3]
```diff
@@ -53,43 +52,51 @@ def run_execution(self, snapshot_id: str, parameters: dict,
...
+                                            project,
+                                        'execution_id':
+                                            previous_execution_id,
+                                        'instance_market_spec':
+                                            instance_market_spec,
+                                        'override_parameters':
+                                            override_parameters,
+                                        'instance_max_uptime_in_minutes':
+                                            instance_max_uptime_in_minutes
+                                    })
```
Comment (SamirTalwar) [inline cli/src/plz/cli/controller_proxy.py]: Perhaps let's set `split_before_first_argument`.

File: cli/src/plz/cli/composition_operation.py [hunk 4]
```diff
@@ -5,21 +5,19 @@
 
 
 class CompositionOperation(Operation):
-    def _run_composition(self,
-                         composition: dict,
+    def _run_composition(self, composition: dict,
```
Comment (SamirTalwar) [inline cli/src/plz/cli/composition_operation.py]: If the line is split, I find it easier to read if it's split on every argument. We can set `split_all_comma_separated_values` to achieve this.

File: cli/src/plz/cli/composition_operation.py [hunk 5]
```diff
@@ -5,21 +5,26 @@
...
+            composition_path: [(str,
+                                Any)]):
         if 'indices_to_compositions' in composition:
             for index, c in composition['indices_to_compositions'].items():
                 self._run_composition(
                     composition=c,
-                    composition_path=composition_path + [
-                        ('parallel', index)])
+                    composition_path=composition_path + [('parallel',
+                                                          index)])
```
Comment (sergiogiro) [inline cli/src/plz/cli/composition_operation.py]: @SamirTalwar  it seems that `split_all_comma_separated_values` triggers things like this (`index)])` fits in the previous line). Do you still want it?

File: cli/src/plz/cli/configuration.py [hunk 6]
```diff
@@ -100,7 +104,9 @@ def _validate_market_spec(configuration, errors, operation: Optional[str]):
 
 
 def _warn_about_instance_max_uptime(
-        configuration, _, operation: Optional[str]):
+        configuration,
+        _,
```
Comment (sergiogiro) [inline cli/src/plz/cli/configuration.py]: And the underscore gets its own line, of course...

File: cli/src/plz/cli/configuration.py [hunk 7]
```diff
@@ -116,49 +122,93 @@ def _warn_about_instance_max_uptime(
...
-            Property('quiet_build', type=bool, default=False),
-            Property('user', required=True),
-            Property('instance_type', default='t2.micro'),
-            Property('project', required=True),
-            Property('image', type=str),
-            Property('image_extensions', type=list, default=[]),
-            Property('command', type=list),
-            Property('input', type=str),
+        prop.name: prop
+        for prop in [
```
Comment (sergiogiro) [inline cli/src/plz/cli/configuration.py]: It gives me shivers...

**Comment by sergiogiro** [comment]
Talked to Samir about SPLIT_ALL_COMMA_SEPARATED_VALUES. It's too agressive. See:
sergiogiro/yapf#1
We want to get that PR merged into yapf and then start using it

**Pull request merged**